subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/report/PageRenderer.java

```diff
@@ -68,13 +69,24 @@ public void doExecute(SimpleHtmlWriter element) throws IOException {
     }
 
     protected void renderFailures(SimpleHtmlWriter htmlWriter) throws IOException {
+        renderTestResultList(htmlWriter, results.getFailures());
+    }
+
+    private void renderTestResultList(SimpleHtmlWriter htmlWriter, Set<TestResult> failures) throws IOException {
         htmlWriter.startElement("ul").attribute("class", "linkList");
-        for (TestResult test : results.getFailures()) {
+        for (TestResult test : failures) {
             htmlWriter.startElement("li");
-            htmlWriter.startElement("a").attribute("href", asHtmlLinkEncoded(getResults().getUrlTo(test.getClassResults()))).characters(test.getClassResults().getSimpleName()).endElement();
+            htmlWriter.startElement("a")
+                .attribute("href", asHtmlLinkEncoded(getResults().getUrlTo(test.getClassResults())))
+                .attribute("title", test.getClassResults().getName())
+                .characters(test.getClassResults().getReportName())
+                .endElement();
             htmlWriter.characters(".");
             String link = asHtmlLinkEncoded(getResults().getUrlTo(test.getClassResults())) + "#" + test.getName();
-            htmlWriter.startElement("a").attribute("href", link).characters(test.getName()).endElement();
+            htmlWriter.startElement("a")
+                .attribute("href", link)
+                .attribute("title", test.getName())
```

Comment:
❌ Why do we use the name as the title? Why not the display name?

Reply:
`title` is something which is not displayed by default - for example, `<a title='title' href='url'>name</a>` will be displayed as `name`, and link to `url`; when you move mouse cursor to the link, it will display `title`. I think `title` matches this scenario: not displayed by default, but user can get it if they want.

Reply:
It's the title of the link. Why would that be the test name instead of the display name? 

The title attribute is used when the link text is very short and you want to gibe the user a better description of what they are clicking on. Using the name would be a worse description.

Reply:
I don't think name is a "better" description, but I do think it's a description which provides more information. I very much hope this can be reserved - do you think we should discard this information at all? Or is there any better place to put them in?

Reply:
I would never look in the tooltip for that information. A tooltip should provide a richer description of what I'm clicking on, not a shorter/less detailed description.

If you think displaying the test method name in addition to the display name is important, then we should display it properly, not hidden somewhere.

Reply:
That's definitely not important, but a "bonus", i.e. if I want to see the exactly class name under certain circumstances, I can get a quick view. If we have to add a hidden attribute to the tag, why can't we leverage the `title` attribute? I'd say let's keep it in `title` attribute.

Reply:
I disagree, but I won't hold up this PR over it.

Maybe @eriwen can give some user experience advise.

Reply:
I don't know if you care, but the information is totally unusable to mobile users.